File: power-square-android/app/src/main/java/com/okla/ops/beans/Cabin.kt
package com.okla.ops.beans

data class Cabin(
    var portNo: Int,//仓门号
    var portName: String,//仓门号展示名
    var batterySn: String,//电池SN
    var batterySoc: Int,//电量
    var chargeCurrent: String,//充电电流
    var status: Int,//0 禁 1 启
    var showStatus: String,//仓门展示状态 例如 启动/禁用
    var batteryStatus: Int,//1 占用 0 空闲
    var showBatteryStatus: String,//仓门电池展示状态 例如 占用/空闲
    var doorStatus: Int,//0 关 1 开
    var showDoorStatus: String,//展示仓门状态 开关
    var swapFlag: Int//电量是否高于阈值 1 是，0 否
)

File: lib/data/models/cabin.dart
import 'package:flutter/foundation.dart';

@immutable
class Cabin {
  final int? portNo;
  final String? portName;
  final String? batterySn;
  final int? status;
  final int? doorStatus;
  final int? lockStatus;
  final int? chargeStatus;
  final int? batterySoc;
  final double? voltage;
  final double? temperature;

  const Cabin({
    this.portNo,
    this.portName,
    this.batterySn,
    this.status,
    this.doorStatus,
    this.lockStatus,
    this.chargeStatus,
    this.batterySoc,
    this.voltage,
    this.temperature,
  });

  factory Cabin.fromJson(Map<String, dynamic> json) {
    return Cabin(
      portNo: (json['portNo'] as num?)?.toInt(),
      portName: json['portName']?.toString(),
      batterySn: json['batterySn']?.toString(),
      status: (json['status'] as num?)?.toInt(),
      doorStatus: (json['doorStatus'] as num?)?.toInt(),
      lockStatus: (json['lockStatus'] as num?)?.toInt(),
      chargeStatus: (json['chargeStatus'] as num?)?.toInt(),
      batterySoc: (json['batterySoc'] as num?)?.toInt(),
      voltage: (json['voltage'] as num?)?.toDouble(),
      temperature: (json['temperature'] as num?)?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() => {
        'portNo': portNo,
        'portName': portName,
        'batterySn': batterySn,
        'status': status,
        'doorStatus': doorStatus,
        'lockStatus': lockStatus,
        'chargeStatus': chargeStatus,
        'batterySoc': batterySoc,
        'voltage': voltage,
        'temperature': temperature,
      };
}
